### ISSUE TITLE
Error in test_get_logits_over_interval (test_utils.TestUtils)

### ISSUE THREAD

**Issue by catherio** [tensorflow/cleverhans#340]
The latest builds (with tf 1.1.0) are failing with the following error:
```
======================================================================
ERROR: test_get_logits_over_interval (test_utils.TestUtils)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/tensorflow/cleverhans/tests_tf/test_utils.py", line 97, in test_get_logits_over_interval
    num_points=num_points)
  File "/home/travis/build/tensorflow/cleverhans/cleverhans/utils.py", line 273, in get_logits_over_interval
    logits = model.get_logits(lin_batch)
  File "/home/travis/build/tensorflow/cleverhans/cleverhans/utils_keras.py", line 151, in get_logits
    logits_name = self._get_logits_name()
  File "/home/travis/build/tensorflow/cleverhans/cleverhans/utils_keras.py", line 141, in _get_logits_name
    node = softmax_layer.inbound_nodes[0]
AttributeError: 'Activation' object has no attribute 'inbound_nodes'
----------------------------------------------------------------------
```

See https://travis-ci.org/tensorflow/cleverhans/jobs/330906885 (build for https://github.com/tensorflow/cleverhans/pull/336)

**Comment by goodfeli** [comment]
Did the version of Keras change recently?
@royaurko any comment?

**Comment by catherio** [comment]
`inbound_nodes` got renamed to `_inbound_nodes` as of https://github.com/keras-team/keras/commit/958239c621a6be037c5f8b30be9270310735f725#diff-fe3ca734e8421a17729d0b0ea4c800ba. What's the right way to proceed?

**Comment by catherio** [comment]
Proposal: replace that line (which is the only use of _inbound_nodes) with a hasattr check and a deprecation warning, to avoid breaking folks who are still using older Keras

**Comment by catherio** [comment]
closed as of #343
